client/my-sites/checkout/checkout-thank-you/marketplace/marketplace-thank-you-theme-section.tsx

```diff
@@ -115,6 +118,10 @@ export const ThankYouThemeSection = ( { theme }: { theme: any } ) => {
 		dispatch( activate( theme.id, siteId, 'marketplace-thank-you' ) );
 	};
 
+	if ( ! isValidThankyouSectionTheme ) {
+		return null;
+	}
```

Comment:
Wouldn't be better to show the theme but with the `activate` button disabled?

Reply:
The idea of this condition is to prevent an invalid scenario where the user manually updates the URL, so I would not add extra logic in the component to handle this flag

Reply:
That's fair, go ahead.